src/fsharp/FSharp.Build/Microsoft.FSharp.NetSdk.targets

```diff
@@ -45,54 +45,24 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
 
   <PropertyGroup Condition="'$(PackProjectInputFile)' == ''">
     <PackProjectInputFile>$(MSBuildProjectFullPath)</PackProjectInputFile>
-  </PropertyGroup>    
-
-  <!--
-    Always reference FSharp.core for coreclr builds
-    TargetFSharpCorePackageIdentity = Name of package to reference      (Default = Microsoft.FSharp.Core.netcore)
-    TargetFSharpCorePackageVersion  = Version of package to reference   (Default = TargetFramework = netcoreapp1.0-1.1 netstandard1.0-1.6 then '1.0.0-*')
-                                                                        (Default = TargetFramework = netcoreapp2.0 netstandard2.0         then '1.0.0-*')
-  -->
-  <PropertyGroup Condition=" '$(TargetFSharpCorePackageVersion)' == '' ">
-    <TargetFSharpCorePackageVersion Condition=" $(TargetFramework.StartsWith(netcoreapp1.)) or $(TargetFramework.StartsWith(netstandard1.)) ">1.0.0-*</TargetFSharpCorePackageVersion>
-    <TargetFSharpCorePackageVersion Condition=" $(TargetFramework.StartsWith(netcoreapp2.)) or $(TargetFramework.StartsWith(netstandard2.)) ">1.0.0-*</TargetFSharpCorePackageVersion>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(TargetFSharpCorePackageIdentity)' == '' ">
-    <TargetFSharpCorePackageIdentity Condition=" '$(TargetFrameworkIdentifier)' != '.NETFramework' " >Microsoft.FSharp.Core.netcore</TargetFSharpCorePackageIdentity>
+  <PropertyGroup Condition=" '$(TargetFSharpCorePackageVersion)' == '' ">
+    <TargetFSharpCorePackageVersion>4.2.0-*</TargetFSharpCorePackageVersion>
```

Comment:
> I wonder what we need from a developer convenience perspective.

It's possible to discuss later as single PR like @dsyme said, is not urgent.

> I expect that we need 4.2.* with an option to enable pre-releases. But i'm not sure what that looks like

I never seen a version range for both .patch (4.2.*) and prerelease.

For me already doesnt make sense that `4.2.0-*` resolve `4.2.0`. 
**Also `4.2.0-rc*` resolve as `4.2.0`**, but reading it, is not what i expect, at all.

But any final choice of desidered range, maybe is just easier to schedule the release of `Fsharp.core` 4.2.0 before the bundling of these target files, so will be easier to fix the range